scalafix-sbt/src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -200,78 +168,34 @@ object ScalafixPlugin extends AutoPlugin {
     }
   )
 
-  // TODO: remove when we can break binary compat
-  lazy val scalafixTaskSettings: Seq[Def.Setting[InputTask[Unit]]] =
-    configureForConfigurations(
-      List(Compile, Test),
-      scalafix,
-      c =>
-        scalafixTaskImpl(
-          c,
-          scalafixParserCompat,
-          compat = true,
-          extraOptions = Nil))
-
-  lazy val scalafixCliTaskSettings: Seq[Def.Setting[InputTask[Unit]]] =
-    configureForConfigurations(
-      List(Compile, Test),
-      scalafixCli,
-      c =>
-        scalafixTaskImpl(c, scalafixParser, compat = false, extraOptions = Nil))
-
-  lazy val scalafixTestTaskSettings: Seq[Def.Setting[InputTask[Unit]]] =
-    configureForConfigurations(
-      List(Compile, Test),
-      scalafixTest,
-      c =>
-        scalafixTaskImpl(
-          c,
-          scalafixParserCompat,
-          compat = true,
-          extraOptions = Seq("--test")))
-
-  @deprecated(
-    "Use configureForConfiguration(List(Compile, Test), ...) instead",
-    "0.5.4")
-  def configureForCompileAndTest(
-      task: InputKey[Unit],
-      impl: Seq[Configuration] => Def.Initialize[InputTask[Unit]]
-  ): Seq[Def.Setting[InputTask[Unit]]] =
-    configureForConfigurations(List(Compile, Test), task, impl)
-
-  /** Configure scalafix/scalafixTest tasks for given configurations */
-  def configureForConfigurations(
-      configurations: Seq[Configuration],
-      task: InputKey[Unit],
-      impl: Seq[Configuration] => Def.Initialize[InputTask[Unit]]
-  ): Seq[Def.Setting[InputTask[Unit]]] =
-    (task := impl(configurations).evaluated) +:
-      configurations.map(c => task.in(c) := impl(Seq(c)).evaluated)
+  lazy val scalafixConfigSettings: Seq[Def.Setting[_]] = scalafixSettings ++
```

Comment:
Can we move this to `autoImport` so users don't have to prefix `ScalafixPlugin.`?